protocol-units/execution/opt-executor/src/executor/services.rs

```diff
@@ -35,49 +35,43 @@ impl Executor {
 
 #[cfg(test)]
 mod tests {
-
 	use super::*;
-	use aptos_crypto::{
-		ed25519::{Ed25519PrivateKey, Ed25519Signature},
-		PrivateKey, Uniform,
-	};
+	use aptos_crypto::{ed25519::Ed25519PrivateKey, Uniform};
 	use aptos_mempool::MempoolClientRequest;
 	use aptos_types::{
-		account_address::AccountAddress,
-		chain_id::ChainId,
-		transaction::{RawTransaction, Script, SignedTransaction, TransactionPayload},
+		account_config, mempool_status::MempoolStatusCode, test_helpers::transaction_test_helpers,
+		transaction::SignedTransaction,
 	};
+	use aptos_vm_genesis::GENESIS_KEYPAIR;
 	use futures::channel::oneshot;
 	use futures::SinkExt;
 	use maptos_execution_util::config::Config;
 
-	fn create_signed_transaction(gas_unit_price: u64, chain_id: ChainId) -> SignedTransaction {
-		let private_key = Ed25519PrivateKey::generate_for_testing();
-		let public_key = private_key.public_key();
-		let transaction_payload = TransactionPayload::Script(Script::new(vec![0], vec![], vec![]));
-		let raw_transaction = RawTransaction::new(
-			AccountAddress::random(),
-			0,
-			transaction_payload,
-			0,
-			gas_unit_price,
-			0,
-			chain_id, // This is the value used in aptos testing code.
-		);
-		SignedTransaction::new(raw_transaction, public_key, Ed25519Signature::dummy_signature())
+	fn create_signed_transaction(
+		sequence_number: u64,
+		maptos_config: &Config,
+	) -> SignedTransaction {
+		let address = account_config::aptos_test_root_address();
+		transaction_test_helpers::get_test_txn_with_chain_id(
+			address,
+			sequence_number,
+			&GENESIS_KEYPAIR.0,
+			GENESIS_KEYPAIR.1.clone(),
+			maptos_config.chain.maptos_chain_id.clone(), // This is the value used in aptos testing code.
+		)
 	}
 
 	#[tokio::test]
 	async fn test_pipe_mempool_while_server_running() -> Result<(), anyhow::Error> {
-		let mut executor = Executor::try_test_default()?;
+		let (mut executor, _tempdir) = Executor::try_test_default(GENESIS_KEYPAIR.0.clone())?;
 		let server_executor = executor.clone();
 
 		let handle = tokio::spawn(async move {
 			server_executor.run_service().await?;
 			Ok(()) as Result<(), anyhow::Error>
 		});
 
-		let user_transaction = create_signed_transaction(0, executor.maptos_config.chain.maptos_chain_id.clone());
+		let user_transaction = create_signed_transaction(1, &executor.maptos_config);
```

Comment:
Do we actually need `1` here?

Reply:
No, 0 works just as well. I have changed this helper function for consistency with the one in the sibling module `transaction_pipe`; perhaps a shared test util module would be better, but we may need different transaction data for different tests, too.

Reply:
Shared is better ofc. But, don't worry about it. Just wanted to clarify conceptually.